## `hw` - homework material

## `lab` - in-class lab material

- `Enrichment_Regions.Rmd` - Peak annotation and functional enrichment

## References

- Stephen B. Baylin and Peter A. Jones, “A Decade of Exploring the Cancer Epigenome — Biological and Translational Implications,” Nature Reviews Cancer 11, no. 10 (September 23, 2011): 726–34, [https://doi.org/10.1038/nrc3130](https://doi.org/10.1038/nrc3130).- Cancer epigenomics introduction, therapies

- Kagohara, Luciane T., Genevieve L. Stein-O’Brien, Dylan Kelley, Emily Flam, Heather C. Wick, Ludmila V. Danilova, Hariharan Easwaran, et al. “Epigenetic Regulation of Gene Expression in Cancer: Techniques, Resources and Analysis.” Briefings in Functional Genomics, August 11, 2017. [https://doi.org/10.1093/bfgp/elx018](https://doi.org/10.1093/bfgp/elx018). - Review of epigeneitc modifications, methylation, histones, chromatin states, 3D. Technologies, databases, software. Lots of references

- Li, Bing, Michael Carey, and Jerry L. Workman. “The Role of Chromatin during Transcription.” Cell 128, no. 4 (February 23, 2007): 707–19. [http://www.cell.com/abstract/S0092-8674(07)00109-2](http://www.cell.com/abstract/S0092-8674(07)00109-2). - Transcription process and the role of chromatin modifications. 

- Zhou, Vicky W., Alon Goren, and Bradley E. Bernstein. “Charting Histone Modifications and the Functional Organization of Mammalian Genomes.” Nature Reviews. Genetics 12, no. 1 (January 2011): 7–18. [https://www.nature.com/articles/nrg2905](https://www.nature.com/articles/nrg2905) - Histone marks review, ChIP-seq. Graphics of histone marks roles

- Wang, Zhibin, Dustin E. Schones, and Keji Zhao. “Characterization of Human Epigenomes.” Current Opinion in Genetics & Development 19, no. 2 (April 2009): 127–34. [https://doi.org/10.1016/j.gde.2009.02.001](https://doi.org/10.1016/j.gde.2009.02.001) - Concise description of main histone marks, their roles in transcription, and the corresponding studies. Figure 2 - schematic distribution of histone marks with respect to genes-TSSs.

- Zhang, Z. D., A. Paccanaro, Y. Fu, S. Weissman, Z. Weng, J. Chang, M. Snyder, and M. B. Gerstein. “Statistical Analysis of the Genomic Distribution and Correlation of Regulatory Elements in the ENCODE Regions.” Genome Research 17, no. 6 (June 1, 2007): 787–97. [https://doi.org/10.1101/gr.5573107](https://doi.org/10.1101/gr.5573107). - ENCODE pilot project analysis. Non-random location of regulatory elements. Enrichment in TSSs, not in the middle or end of transcription sites. PCA and biplot representation of interrelatedness among TFs and histone marks, clustering.

- Fu, Audrey Qiuyan, and Boris Adryan. “Scoring Overlapping and Adjacent Signals from Genome-Wide ChIP and DamID Assays.” Molecular BioSystems 5, no. 12 (December 2009): 1429–38. [http://pubs.rsc.org/en/content/articlelanding/2009/mb/b906880e](http://pubs.rsc.org/en/content/articlelanding/2009/mb/b906880e)

- Huen, David S., and Steven Russell. “On the Use of Resampling Tests for Evaluating Statistical Significance of Binding-Site Co-Occurrence.” BMC Bioinformatics 11 (June 30, 2010): 359. [https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-359](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-359) - Thorough review on permutation. Problem with different numbers of ROIs/epiregions in permutations. Regions may overlap (i.e., occur in clusters). Independent assignment is best.

- McLean, Cory Y., Dave Bristor, Michael Hiller, Shoa L. Clarke, Bruce T. Schaar, Craig B. Lowe, Aaron M. Wenger, and Gill Bejerano. “GREAT Improves Functional Interpretation of Cis-Regulatory Regions.” Nature Biotechnology 28, no. 5 (May 2010): 495–501. [https://doi.org/10.1038/nbt.1630](https://doi.org/10.1038/nbt.1630). - Hypergeometric and binomial enrichment of regulatory regions in relation to genesgenomic regions and their ontologies.

- Dozmorov, Mikhail G. “Epigenomic Annotation-Based Interpretation of Genomic Data: From Enrichment Analysis to Machine Learning.” Bioinformatics (Oxford, England) 33, no. 20 (October 15, 2017): 3323–30. [https://doi.org/10.1093/bioinformatics/btx414](https://doi.org/10.1093/bioinformatics/btx414)

- Eddy, Sean R. “What Is a Hidden Markov Model?” Nature Biotechnology 22, no. 10 (October 2004): 1315–16. [https://doi.org/10.1038/nbt1004-1315](https://doi.org/10.1038/nbt1004-1315)

- Eddy, S. R. “Multiple Alignment Using Hidden Markov Models.” Proceedings. International Conference on Intelligent Systems for Molecular Biology 3 (1995): 114–20. [https://pdfs.semanticscholar.org/bc1e/ffe17026c1396697e4ea2b399f7a049202fc.pdf](https://pdfs.semanticscholar.org/bc1e/ffe17026c1396697e4ea2b399f7a049202fc.pdf) - HMMER hidden markov models for protein sequence alignment. First paper

- Schuster-Böckler, Benjamin, and Alex Bateman. “An Introduction to Hidden Markov Models.” In Current Protocols in Bioinformatics, edited by Andreas D. Baxevanis, Daniel B. Davison, Roderic D.M. Page, Gregory A. Petsko, Lincoln D. Stein, and Gary D. Stormo. Hoboken, NJ, USA: John Wiley & Sons, Inc., 2007. [https://doi.org/10.1002/0471250953.bia03as18](https://doi.org/10.1002/0471250953.bia03as18) - Hidden Markov Models primer

- L.R. Rabiner, “A Tutorial on Hidden Markov Models and Selected Applications in Speech Recognition,” Proceedings of the IEEE 77, no. 2 (February 1989): 257–86, [https://doi.org/10.1109/5.18626](https://doi.org/10.1109/5.18626) - Theory and statistics of Hidden Markov Models. Very detailed, thorough

- "Hidden Markov Models" by bioinformaticsalgorithms.com. [https://www.youtube.com/watch?list=PLQ-85lQlPqFPnk31Uut2ajVkBvlFmMtdx&v=cnvYujS9Rlk](https://www.youtube.com/watch?list=PLQ-85lQlPqFPnk31Uut2ajVkBvlFmMtdx&v=cnvYujS9Rlk)

- Ernst, Jason, and Manolis Kellis. “Chromatin-State Discovery and Genome Annotation with ChromHMM.” Nature Protocols 12, no. 12 (December 2017): 2478–92. [https://doi.org/10.1038/nprot.2017.124](https://doi.org/10.1038/nprot.2017.124) - ChromHMM protocol. Intro about ChromHMM, other methods. Links to genome annotation models.

- Hoffman, Michael M., Orion J. Buske, Jie Wang, Zhiping Weng, Jeff A. Bilmes, and William Stafford Noble. “Unsupervised Pattern Discovery in Human Chromatin Structure through Genomic Segmentation.” Nature Methods 9, no. 5 (May 2012): 473–76. [https://doi.org/10.1038/nmeth.1937](https://doi.org/10.1038/nmeth.1937) - Segway - segmentation and prediction of genomic states. Using dynamic Bayesian network


## `misc` - misc presentations and materials

- `hidden_markov_models.pdf` - Hidden Markov Models. Ben Langmead. http://www.cs.jhu.edu/~langmea/resources/lecture_notes/hidden_markov_models.pdf

- `lecture_17.pdf` - Hidden Markov Model reading. https://ocw.mit.edu/courses/mathematics/18-417-introduction-to-computational-molecular-biology-fall-2004/lecture-notes/lecture_17.pdf

- `markov_chains.pdf` - Markov chain for CpG discovery. Ben Langmead, 1st lec http://www.cs.jhu.edu/~langmea/resources/lecture_notes/markov_chains.pdf

- `MIT7_91JS14_Lecture10.pdf` - Markov & Hidden Markov Models of Genomic & Protein Features. https://ocw.mit.edu/courses/biology/7-91j-foundations-of-computational-and-systems-biology-spring-2014/lecture-slides/MIT7_91JS14_Lecture10.pdf, https://www.youtube.com/watch?v=d5NMrA2HkG4

## ToDo - todo list

- Segway dynamic Bayesian network explanation - ~15min https://www.youtube.com/watch?v=iKLvCuFD1MA&list=PLUl4u3cNGP63uK-oWiLgO7LLJV6ZCWXac&index=18, and at Hoffman, Michael M., Orion J. Buske, Jie Wang, Zhiping Weng, Jeff A. Bilmes, and William Stafford Noble. “Unsupervised Pattern Discovery in Human Chromatin Structure through Genomic Segmentation.” Nature Methods 9, no. 5 (May 2012): 473–76. doi:10.1038/nmeth.1937.




