---
categories:
 - weekly
title: "11 - Metagenomics"
slides: ""
tags:
 - assets
visible: 1
---

[Slides: Metagenomics]({{site.baseurl}}/assets/11_Metagenomics/01_Metagenomics.pdf)  

## References

- Ghodsi, Mohammadreza, Bo Liu, and Mihai Pop. “DNACLUST: Accurate and Efficient Clustering of Phylogenetic Marker Genes.” BMC Bioinformatics 12 (June 30, 2011): 271. [https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-271](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-271) DNACLUST - metagenomics clustering of 16S sequencing. Comparison with CD-HIT and UCLUST. Recruit sequences within the radius around cluster seed. Explanation of distance, Needleman-Wunsch algorithm.

- Grice, Elizabeth A., and Julia A. Segre. “The Human Microbiome: Our Second Genome.” Annual Review of Genomics and Human Genetics 13, no. 1 (September 22, 2012): 151–70. [https://www.annualreviews.org/doi/abs/10.1146/annurev-genom-090711-163814](https://www.annualreviews.org/doi/abs/10.1146/annurev-genom-090711-163814) - Review on microbiome, 16S RNA, tools, human microbiome project.

- Kong, Heidi H. “Skin Microbiome: Genomics-Based Insights into the Diversity and Role of Skin Microbes.” Trends in Molecular Medicine 17, no. 6 (June 2011): 320–28. https://doi.org/10.1016/j.molmed.2011.01.013. - Skin microbiome review. Introduction about microbiome sequencing (culture and direct), 16S sequencing, other markers. Definitions. Skin-specific findings.

- McDonald, Daniel, Zhenjiang Xu, Embriette R. Hyde, and Rob Knight. “Ribosomal RNA, the Lens into Life.” RNA 21, no. 4 (April 2015): 692–94. [http://rnajournal.cshlp.org/content/21/4/692.full](http://rnajournal.cshlp.org/content/21/4/692.full) - Short review of 16S rRNA history, databases (RDP< Greengenes, SILVA), QIIME

- Huson, D. H., A. F. Auch, J. Qi, and S. C. Schuster. “MEGAN Analysis of Metagenomic Data.” Genome Research 17, no. 3 (February 6, 2007): 377–86. [https://genome.cshlp.org/content/17/3/377.abstract](https://genome.cshlp.org/content/17/3/377.abstract) - MEGAN paper. Intro into metagenomics, sequencing, alignment analysis. LCA (lowest common ancestor) algorithm.

- Kuczynski, Justin, Christian L. Lauber, William A. Walters, Laura Wegener Parfrey, José C. Clemente, Dirk Gevers, and Rob Knight. “Experimental and Analytical Tools for Studying the Human Microbiome.” Nature Reviews Genetics 13, no. 1 (December 16, 2011): 47–58. [https://www.nature.com/articles/nrg3129](https://www.nature.com/articles/nrg3129)

- Hamady, Micah, and Rob Knight. “Microbial Community Profiling for Human Microbiome Projects: Tools, Techniques, and Challenges.” Genome Research 19, no. 7 (July 2009): 1141–52. [http://genome.cshlp.org/cgi/pmidlookup?view=long&pmid=19383763](http://genome.cshlp.org/cgi/pmidlookup?view=long&pmid=19383763) - Introduction, background of metagenomics. 16S vs. sequencing, experimental questions like read length, sampling depth, how to analyze.

- Schloss, P. D., S. L. Westcott, T. Ryabin, J. R. Hall, M. Hartmann, E. B. Hollister, R. A. Lesniewski, et al. “Introducing Mothur: Open-Source, Platform-Independent, Community-Supported Software for Describing and Comparing Microbial Communities.” Applied and Environmental Microbiology 75, no. 23 (December 1, 2009): 7537–41. [https://doi.org/10.1128/AEM.01541-09](https://doi.org/10.1128/AEM.01541-09) - mothur paper. [https://www.mothur.org/](https://www.mothur.org/)

- Wang, Q., G. M. Garrity, J. M. Tiedje, and J. R. Cole. “Naive Bayesian Classifier for Rapid Assignment of RRNA Sequences into the New Bacterial Taxonomy.” Applied and Environmental Microbiology 73, no. 16 (August 15, 2007): 5261–67. [https://doi.org/10.1128/AEM.00062-07](https://doi.org/10.1128/AEM.00062-07) - Naive Bayes classifier used in Ribosome Database Project (RDP). Good methods statistical description.

- Sczyrba, Alexander, Peter Hofmann, Peter Belmann, David Koslicki, Stefan Janssen, Johannes Dröge, Ivan Gregor, et al. “Critical Assessment of Metagenome Interpretation-a Benchmark of Metagenomics Software.” Nature Methods 14, no. 11 (November 2017): 1063–71. [https://doi.org/10.1038/nmeth.4458](https://doi.org/10.1038/nmeth.4458) - Critical Assessment of Metagenome Interpretation (CAMI) paper. Assessment of microbial genome assemblers, effect of sequencing depth, strain diversity, taxonomic binning, etc., recommendations on software and best practices.

### Metagenomic resources

- List of software packages (and the people developing these methods) for microbiome (16S), metagenomics (WGS, Shot-gun sequencing), and pathogen identification/detection/characterization. [https://github.com/stevetsa/awesome-microbes](https://github.com/stevetsa/awesome-microbes)

- "Analysis of Metagenomic Data (2016)" course. [https://bioinformatics-ca.github.io/analysis_of_metagenomic_data_2016/](https://bioinformatics-ca.github.io/analysis_of_metagenomic_data_2016/)

- Metagenomic Assembly Workshop, [https://2014-5-metagenomics-workshop.readthedocs.io/en/latest/index.html](https://2014-5-metagenomics-workshop.readthedocs.io/en/latest/index.html)