- `01_File_formats.Rmd` - genomic file formats
- `02_Genomic_Resources.Rmd`
- `03_TCGA.Rmd`

## `lab` - in-class lab material

- `04_Genomic_Resources_tutorial_Quinlan.pdf` - step-by-step tutorial on UCSC GB and Ensembl, by Aaron Quinlan, from https://github.com/quinlan-lab/applied-computational-genomics

# ToDo

# ToRead

- Schroeder, Michael P., Abel Gonzalez-Perez, and Nuria Lopez-Bigas. “Visualizing Multidimensional Cancer Genomics Data.” Genome Medicine 5, no. 1 (2013): 9. https://doi.org/10.1186/gm413. - Omics visualization tools review, summary table.

- Silva, Tiago C., Antonio Colaprico, Catharina Olsen, Fulvio D’Angelo, Gianluca Bontempi, Michele Ceccarelli, and Houtan Noushmehr. “TCGA Workflow: Analyze Cancer Genomics and Epigenomics Data Using Bioconductor Packages.” F1000Research 5 (December 28, 2016): 1542. https://doi.org/10.12688/f1000research.8923.2. - TCGA workflow

- Colaprico, Antonio, Tiago C. Silva, Catharina Olsen, Luciano Garofano, Claudia Cava, Davide Garolini, Thais S. Sabedot, et al. “TCGAbiolinks: An R/Bioconductor Package for Integrative Analysis of TCGA Data.” Nucleic Acids Research 44, no. 8 (May 5, 2016): e71. https://doi.org/10.1093/nar/gkv1507.

- Cline, Melissa S., Brian Craft, Teresa Swatloski, Mary Goldman, Singer Ma, David Haussler, and Jingchun Zhu. “Exploring TCGA Pan-Cancer Data at the UCSC Cancer Genomics Browser.” Scientific Reports 3 (October 2, 2013): 2652. https://doi.org/10.1038/srep02652.

- Collado-Torres, Leonardo, Abhinav Nellore, Kai Kammers, Shannon E Ellis, Margaret A Taub, Kasper D Hansen, Andrew E Jaffe, Ben Langmead, and Jeffrey T Leek. “Reproducible RNA-Seq Analysis Using Recount2.” Nature Biotechnology 35, no. 4 (April 11, 2017): 319–21. https://doi.org/10.1038/nbt.3838. - recount2 - uniformly processed RNA-seq data, using Rail-RNA. Counts and bigWig files for differential region analysis using derfinder.

- Parker, Joel S., Michael Mullins, Maggie C. U. Cheang, Samuel Leung, David Voduc, Tammi Vickery, Sherri Davies, et al. “Supervised Risk Predictor of Breast Cancer Based on Intrinsic Subtypes.” Journal of Clinical Oncology: Official Journal of the American Society of Clinical Oncology 27, no. 8 (March 10, 2009): 1160–67. https://doi.org/10.1200/JCO.2008.18.1370. - PAM50 paper. BRCA classification into Luminal A, Luminal B, HER2-enriched, basal-like and normal-like subtypes. 50 genes determined by Prediction Analysis of Microarrays, remain significant predictors in uni- and multivariate analyses. Data links at https://www.biostars.org/p/77590/

### Workflows and tools

- UCSC genome browser video tutorials: [http://genome.ucsc.edu/training/vids/](http://genome.ucsc.edu/training/vids/)

- UCSC tutorial, common data formats, Galaxy tutorial, mentioning of VEP, IGV. [http://web.cse.ohio-state.edu/~machiraju.1/teaching/CSE5599-BMI7830/Lectures/pdf/BMI7830-CSE5559-2014-huang-machiraju-Week-11-1.pdf](http://web.cse.ohio-state.edu/~machiraju.1/teaching/CSE5599-BMI7830/Lectures/pdf/BMI7830-CSE5559-2014-huang-machiraju-Week-11-1.pdf)

- IGV tutorial, [https://github.com/griffithlab/rnaseq_tutorial/wiki/IGV-Tutorial](https://github.com/griffithlab/rnaseq_tutorial/wiki/IGV-Tutorial)

- Analyze cancer genomics and epigenomics data using Bioconductor packages, [https://github.com/BioinformaticsFMRP/TCGAWorkflow/](https://github.com/BioinformaticsFMRP/TCGAWorkflow/), and the associated paper - Silva, Tiago C., Antonio Colaprico, Catharina Olsen, Fulvio D’Angelo, Gianluca Bontempi, Michele Ceccarelli, and Houtan Noushmehr. “TCGA Workflow: Analyze Cancer Genomics and Epigenomics Data Using Bioconductor Packages.” F1000Research 5 (December 28, 2016): 1542. https://doi.org/10.12688/f1000research.8923.2.

- Tutorial for the sevenbridges cancer genomics cloud, [https://github.com/tdelhomme/CancerGenomicsCloud_tutorial](https://github.com/tdelhomme/CancerGenomicsCloud_tutorial)

- "Read GTF file into R", GTF intro, tutorial for command line and R handling of GTF info, https://davetang.org/muse/2017/08/04/read-gtf-file-r/

- VCF-tips-tricks, [https://github.com/tdelhomme/VCF-tips-tricks](https://github.com/tdelhomme/VCF-tips-tricks)

# `misc`

- `ngsplot_Apr2014.pdf` - Visualization of Next-Gen Sequencing (NGS) Data: ngsplot, http://barc.wi.mit.edu/education/hot_topics/ngsplot/ngsplot_Apr2014.pdf
- Data - http://barc.wi.mit.edu/education/hot_topics/ngsplot/ngsplot_demo.zip, ~500Mb


# `ToDo`

- `ngsplot` - Quick mining and visualization of NGS data by integrating genomic databases. https://github.com/shenlab-sinai/ngsplot