---
layout: page
title: Syllabus, "Statistical Methods for High-throughput Genomic Data II" course, BIOS 668
description: Syllabus for the "Statistical Methods for High-throughput Genomic Data II" course BIOS 668
---

**Instructor**: 	Mikhail G. Dozmorov, Ph.D.

**Schedule**: 		Monday, Wednesday 9:00am – 10:20am

**Classroom**: 		One Capital Square (OCS) 5009

**Office hours**: 	Mon, Wed 10:30am – 12:00pm, OCS 730

**Prerequisites**:   BIOS 567 Statistical Methods for High-throughput Genomic Data I or permission by the instructor.

**Required course material**: posted via Blackboard and on the course' web site https://mdozmorov.github.io/BIOS668.2018/

**Software**: 	Unix, R programming environment, GitHub

# Course description

The study of genomics and use of next-generation sequencing are at the forefront of biomedical research. Sequencing market is constantly evolving, stimulating the development of new analytical approaches and software tools. Therefore it may not be possible to maintain a stable analysis pipeline throughout a project because the lifetime of software often spans months and even years. To be able to effectively analyze and interpret genomic sequencing data, it is crucial to (1) understand the technologies that produce the data and (2) develop strong computational skills that will be flexible in this dynamic environment.

This course is a continuation of the BIOS 567 and will introduce high-throughput genomic assays including DNA sequencing and genome variation analysis, transcriptome profiling with RNA-seq and miRNA-seq, metagenomics, epigenomic analysis including ChIP-seq and methylation assays, single-cell sequencing, and chromatin conformation capture technologies. The course is primarily focusing on human genomics; however, knowledge and skills gained through the course are extendable on genomics of model organisms.

# Course Objectives

* Understand the core principles, strengths, and limitations of high-throughput technologies
* Learn statistical models and approaches used in sequencing data analysis
* Gain practical experience in high-throughput Exploratory Data Analysis, visualization, and quality control using Unix and R environment
* Critically evaluate and interpret statistical methods used in flagship tools for sequencing data analysis
* Interpret biological findings provided by different sequencing technologies, and be able to integrate different layers of -omics data

# Grading

* Class participation (20%)
* Reading and homework assignment (50% grade composition). Homework is due two weeks from the date of assignment, unless specified otherwise. Late homework assignments will not receive any credit, unless discussed with the instructor
* Problem solving, programming assignments, writing, oral presentations 
* Final project (30%)
* The following scale is strictly observed (no rounding): A: 90-100, B: 80-89, C: <80

Homework and other assignments should be submitted as reproducible reports using R/markdown. The final project should be submitted as a GitHub repository.

# Class Rules

* Attendance is required
* Read all assignments before class
* Bring your laptop and the book to every class
* Observe the [VCU Honor Pledge](https://students.vcu.edu/studentconduct/vcu-honor-system/academic-misconduct-/honor-pledge/) in any class- and homework activities

# Course topics

-  Unix overview
-  Genomic technologies and Bioconductor
-  Genome sequencing and alignment
-  Single nucleotide polymorphism (SNP) analysis
-  Copy Number Variant (CNV) analysi
-  RNA- and miRNA-seq
-  Chromatin Immunoprecipitation (ChIP)-, DNAse-, ATAC-seq
-  Metagenomic
-  Methylation
-  Epigenomics
-  Chromatin Conformation Capture analysis
-  Single-cell sequencin
-  Integrative analysis

<!--[Final project](final_project.html)-->


[University-wide policies](http://www.provost.vcu.edu/academic-affairs/operations/syllabus-statements/)